assets/src/editor/events/ticket-assignments-manager/filters/with-ticket-assignments-filters.js

```diff
@@ -0,0 +1,156 @@
+import filterReducer from './reducer';
+
+const { FormRow } = twoColumnAdminFormLayout;
+
+/*
+ * Enhanced TicketAssignments with filters for
+ * toggling display of archived and expired entities
+ */
+const withTicketAssignmentsFilters = createHigherOrderComponent(
+	( WrappedComponent ) => ( {
+		dateEntity,
+		dateEntities,
+		ticketEntity,
+		ticketEntities,
+		...otherProps
+	} ) => {
+		const [ filterState, setFilter ] = useReducer(
+			filterReducer,
+			DEFAULT_FILTER_STATE
+		);
+		const {
+			showArchivedDates,
+			showExpiredDates,
+			showArchivedTickets,
+			showExpiredTickets,
+		} = filterState;
+		const unfilteredDatesCount = dateEntities.length;
+		const unfilteredTicketCount = ticketEntities.length;
+		const filterDateEntities = useCallback(
+			() => dateEntity ?
+				[ dateEntity ] :
+				filterDates(
+					dateEntities,
+					showArchivedDates,
+					showExpiredDates
+				),
+			[ dateEntity, dateEntities, showArchivedDates, showExpiredDates ]
+		);
+		const filterTicketEntities = useCallback(
+			() => ticketEntity ?
+				[ ticketEntity ] :
+				filterTickets(
+					ticketEntities,
+					showArchivedTickets,
+					showExpiredTickets
+				),
+			[
+				ticketEntity,
+				ticketEntities,
+				showArchivedTickets,
+				showExpiredTickets,
+			]
+		);
+		const filteredDateEntities = filterDateEntities();
+		const filteredTicketEntities = filterTicketEntities();
+		const showDateFilters = useCallback(
+			() => getShowDateFilters(
+				dateEntity,
+				filteredDateEntities.length,
+				unfilteredDatesCount,
+				showArchivedDates,
+				showExpiredDates
+			),
+			[
+				dateEntity,
+				filteredDateEntities,
+				unfilteredDatesCount,
+				showArchivedDates,
+				showExpiredDates,
+			]
+		);
+		const showTicketFilters = useCallback(
+			() => getShowTicketFilters(
+				ticketEntity,
+				filteredTicketEntities.length,
+				unfilteredTicketCount,
+				showArchivedTickets,
+				showExpiredTickets
+			),
+			[
+				ticketEntity,
+				filteredTicketEntities,
+				unfilteredTicketCount,
+				showArchivedTickets,
+				showExpiredTickets,
+			]
+		);
+		const ticketAssignmentsFilters = (
+			<Fragment>
+				<FormRow>
+					<DateFilters
+						showArchivedDates={ showArchivedDates }
+						showExpiredDates={ showExpiredDates }
+						toggleArchivedDates={
+							() => setFilter( showArchivedDatesAction )
+						}
+						toggleExpiredDates={
+							() => setFilter( showExpiredDatesAction )
+						}
+						showDateFilters={ showDateFilters() }
+						showTicketFilters={ showTicketFilters() }
+					/>
+					<TicketFilters
+						showArchivedTickets={ showArchivedTickets }
+						showExpiredTickets={ showExpiredTickets }
+						toggleArchivedTickets={
+							() => setFilter( showArchivedTicketsAction )
+						}
+						toggleExpiredTickets={
+							() => setFilter( showExpiredTicketsAction )
+						}
+						showDateFilters={ showDateFilters() }
+						showTicketFilters={ showTicketFilters() }
```

Comment:
hmm better, but I was still thinking we could reduce the amount of props passes through to the Filter components.  Think of cases where there may be additional filter state passed through (right now the Filter components are hardcoded to only support toggling archived or expired state)?  For now, in the interest of keeping momentum and getting this pull merged we can just use it as is as I don't have the energy at the moment to propose alternatives 😄

Reply:
There's some interplay between the different filters so I've had no choice but to "lift state" to the highest component in the tree (or lowest? ie: closer to trunk vs top of tree!?!) so at some point we have to pass the props down. 

The only way to reduce things more would be to bundle the props into an object and pass that, but imho that's not good because it hides the implementation details / component signature and potentially interferes with memoization.

I think the biggest win with this branch is the fact that the filters are at least all self contained and the only things being passed to `TicketAssignmentsManagerModal` is the new `withTicketAssignmentsFilters` HOC and a single prop for displaying the filters. If we need to add more filters or completely change their configuration or whatever, the effects on TAM will be minimal if anything at all.